src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -64,7 +64,7 @@ private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccount
 
             BigInteger gas = CalculateBonus(engine.Snapshot, state.VoteTo, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
             state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
-            GAS.Mint(engine, account, gas);
+            GAS.Mint(engine, account, gas, false);
```

Comment:
This is the most interesting thing here and I'm really not sure about it. There might be scenarios when NEO-holding contract would like to do something with generated GAS and this makes it less convenient, furthermore this can effectively lock GAS generated into contract's address. Maybe we should have `true` here and leave this as contract's responsibility (if it accepts NEO, it must handle this minted GAS in some way also).

Reply:
I think that GAS mint should be avoided of this logic

Reply:
> if it accepts NEO, it must handle this minted GAS in some way

I agree

Reply:
Maybe this should be `true`.

Reply:
What should you do if you want to receive NEO, and not GAS?

Reply:
Check `from`? If it's not minted (non-null), reject it, if it is --- deal with it.

Reply:
But every neo transaction, mint gas

Reply:
In most cases, it's non-contract address.